### ISSUE TITLE
[RNN] TFLite does not appear to be using the UnidirectionalSequenceLSTM

### ISSUE THREAD

**Issue by eric** [tensorflow/tensorflow#73254]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS
- TensorFlow installation (pip package or built from source): from pip
- TensorFlow library (version, if pip package or github SHA, if built from source): keras-nightly-3.4.1.dev2024080603 tb-nightly-2.18.0a20240806 tf-nightly-2.18.0.dev20240806

### 2. Code

1) https://colab.research.google.com/gist/eric/f00f071e527f9fa7b2ed39f8d482fbb4/tensorflow-datasets.ipynb
2) https://colab.research.google.com/gist/eric/a292799568831371b7686a2b8cefcd0b/tensorflow-lite-debugger-colab.ipynb

[model.zip](https://github.com/user-attachments/files/16516496/model.zip)

### 3. Failure after conversion

I've found that any Keras LSTM causes this issue.

I expected TFLite to use the `UnidirectionalSequenceLSTM` op, but instead it seems to be doing something else that then requires the use of flex ops, which I would like to avoid trying to get to work with my tflite deployment situation.

**Comment by eric** [comment]
Is there anything I can do to tweak my model to allow it to work without the flex ops?

**Comment by pkgoogle** [comment]
Hi @eric, using only TFLite BuiltIn Ops, will do this but that can be quite constraining. For most LSTM use cases we highly recommend upgrading to Transformers and using PyTorch and [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch) and the [Generative API](https://github.com/google-ai-edge/ai-edge-torch?tab=readme-ov-file#generative-api) to convert to .tflite models. Does that work for you?

**Comment by eric** [comment]
So far I've found that using an LSTM has been a good trade-off for performance and accuracy, so it would be nice if they could work again. Is there a current working example that shows how to use it with tensorflow?

**Comment by pkgoogle** [comment]
Hi @eric, with raw tensorflow w/o Keras, it does not seem like there is an easy way to create an LSTM. It was sunsetted awhile ago ... I did find a raw_ops for an LSTM Cell: https://www.tensorflow.org/api_docs/python/tf/raw_ops/BlockLSTMV2. I don't believe you will want to reconstruct an LSTM with more primitive ops w/o Keras but that is one possibility.

**Comment by eric** [comment]
Is there a way to do it with Keras? I am using Keras so that would  be fine.

**Comment by pkgoogle** [comment]
Hi @eric, my current attempts seems to be running into a prior bug ... would you be willing to use [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch) instead?

I was able to convert an lstm that way:
```py
import ai_edge_torch
import torch
import torch.nn as nn


lstm = nn.LSTM(10, 20, 2)
h0 = torch.randn(2, 3, 20)
c0 = torch.randn(2, 3, 20)

sample_input = (torch.randn(5, 3, 10), (h0, c0))

edge_model = ai_edge_torch.convert(lstm.eval(), sample_input)
edge_model.export("lstm.tflite")
```

**Comment by eric** [comment]
I've been using keras-tcn for other parts of my model and found that it performed better than LSTM, so I would have to evaluate the pytorch landscape to see if it would work for me.

**Comment by pkgoogle** [comment]
Understood, let us know if for some reason that workflow doesn't work. You are able to reopen the issue in the future.

**Comment by eric** [comment]
Is there any hope of fixing the bug in Keras that is preventing it from working? It would be nice to not have to rewrite everything.

**Comment by pkgoogle** [comment]
Hi @eric, we can turn this issue into that issue for now, if that is preferred. Here's a gist showing the issue with tf-nightly (I have attempted on previous versions of tf up to and including 2.15) [gist](https://colab.sandbox.google.com/gist/pkgoogle/71316c19106db05fc455913dea6cf366/73254.ipynb)

**Comment by eric** [comment]
That would be great. Thanks!

**Comment by gaikwadrahul8** [comment]
Hi, @eric 
Thanks for raising this issue. Are you aware of the migration to [LiteRT](https://developers.googleblog.com/en/tensorflow-lite-is-now-litert/)? This transition is aimed at enhancing our project's capabilities and providing improved support and focus for our users. As we believe this issue is still relevant to LiteRT we are moving your issue there. Please follow progress here: https://github.com/google-ai-edge/LiteRT/issues/41

Let us know if you have any questions. Thanks.
